DataFormats/TotemReco/src/classes_def.xml

```diff
@@ -7,4 +7,12 @@
   <class name="edm::Wrapper<edm::DetSetVector<TotemT2Digi> >"/>
   <class name="std::vector<TotemT2Digi>"/>
   <class name="std::vector<edm::DetSet<TotemT2Digi> >"/>
+  <class name="TotemT2RecHit" ClassVersion="2">
+    <version ClassVersion="2" checksum="969691996"/>
+  </class>
+  <class name="edm::DetSet<TotemT2RecHit>"/>
+  <class name="edm::DetSetVector<TotemT2RecHit>"/>
```

Comment:
> 1\. would the API change for all subsequent consumer code using this format?

It likely would, depending on how the `edm::DetSetVector` is being used. IIRC just iterating over the elements is not much different, but some other access patterns could be more different (or not available).

> 2\. would a migration campaign from `edm::DetSetVector<T>` to `edmNew::DetSetVector<T>` be necessary for all PPS objects in `DataFormats/CTPPSReco` and `DataFormats/CTPPSDigi`? 

If such a change would be made, yes, it would be good to replace all uses of `edm::DetSetVector` with `edmNew::DetSetVector` (and that would certainly be beyond this PR). Unfortunately it would also break backwards compatibility, which means such the moment for such a change would need some planning.

> Alternatively, is there any plan to "simply" replace `edm::DetSetVector<T>` by its `edmNew::DetSetVector<T>` (with all the conversion codes it implies)?

We have long hoped the `edm::DetSetVector` could eventually be phased out (#22711), but have not taken any real action towards that (lower hanging fruits, people usually get annoyed by data format backwards incompatibilities). But `edmNew::DetSetVector` is not (and can't really be made) as a drop-in replacement of `edm::DetSetVector` (not sure if you meant this, but just to make sure).

Reply:
Indeed, I can tell these access patterns are much different (and would require an overall improvement in the API to be made convenient IMHO). In order not to add a few unnecessary signatures in the heat of the Summer holidays I will also replace the `edm::DetSetVector<TotemT2Digi>` collection by a `edmNew::DetSetVector<TotemT2Digi>` in a follow-up PR.